src/rdkafka_request.c

```diff
@@ -219,6 +219,7 @@ rd_kafka_topic_partition_list_t *rd_kafka_buf_read_topic_partitions(
         int32_t TopicArrayCnt;
         rd_kafka_topic_partition_list_t *parts = NULL;
 
+        // TODO: check topic array to be null case.
```

Comment:
I think this is the generic function and should take care of Null. It can't be null through KIP-848 for sure as we check whether the assignment are present or not first. I would update it to be a FIXME maybe.

Reply:
Updating to include both.